### PR TITLE
Fix documentation

### PR DESCRIPTION
**Pull request by ptormene**
Fixes https://github.com/gem/oq-engine/issues/8629

I've fixed some broken cross-references, a malformed table, some indentation issues and various details.
I've added the parameter `-W` to sphinx-build, to let the build fail in case of any error or warning. Unfortunately I couldn't find any way to let the build produce an exit code different from 0 in case of errors (without `-W` also errors are ignored, not only warnings).
Sphinx was complaining about `language = None` and it was falling back to `'en'`. I made it explicit. Please @nastasi-oq check if it was not set on purpose.

### CODE REVIEW

**Comment by micheles** [comment]
There was a bug with the exit code of sphinx 6 years ago, but they say it is fixed: https://github.com/sphinx-doc/sphinx/pull/4160 (?)

File: doc/sphinx/openquake.hazardlib.rst [hunk 1]
```diff
@@ -12,8 +12,6 @@ Subpackages
     openquake.hazardlib.mfd
     openquake.hazardlib.scalerel
     openquake.hazardlib.source
-    openquake.hazardlib.source_reader
-    openquake.hazardlib.logictree
```
Comment (micheles) [inline doc/sphinx/openquake.hazardlib.rst:16]: This 2 lines should not be removed. Instead openquake.commonlib.source_reader and openquake.commonlib.logictree should be removed.

Comment (ptormene) [reply doc/sphinx/openquake.hazardlib.rst:16]: They are already present below as automodules

**Comment by ptormene** [comment]
I've found also this discussion from April 2022: https://github.com/sphinx-doc/sphinx/issues/10397
So it looks like the problem of the wrong status code was not fixed 6 years ago.

**Review by micheles — APPROVED** [review]

**Comment by ptormene** [comment]
Now I have this green: https://github.com/gem/oq-engine/actions/runs/5133991839/jobs/9237350113

**Pull request merged**